File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/guild/tax/AdditionalTaxCollectorInformations.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait AdditionalTaxCollectorInformations extends TypeDeserializer {
  val typeId = 165

  protected trait Type extends BaseType with Serializable {
    val typeId = AdditionalTaxCollectorInformations.typeId

    val collectorCallerName: String
    val date: Int

    def serialize(buf: Buffer) {
      buf.writeString(collectorCallerName)
      buf.writeInt(date)
    }

  }

  protected def create(collectorCallerName: String, date: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val collectorCallerName = buf.readString()
    val date = buf.readInt()

    create(collectorCallerName, date)
  }

}

object AdditionalTaxCollectorInformations extends AdditionalTaxCollectorInformations {
  case class TypeImpl(collectorCallerName: String, date: Int) extends Type
  override def create(collectorCallerName: String, date: Int) = TypeImpl(collectorCallerName, date)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(collectorCallerName: String, date: Int) = create(collectorCallerName, date)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/AdditionalTaxCollectorInformations.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AdditionalTaxCollectorInformations extends NetworkType {
    public static final short TYPE_ID = 165;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public String collectorCallerName;
    public int date;
    
    public AdditionalTaxCollectorInformations() { }
    
    public AdditionalTaxCollectorInformations(String collectorCallerName, int date) {
        this.collectorCallerName = collectorCallerName;
        this.date = date;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(collectorCallerName);
        buf.writeInt(date);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        collectorCallerName = buf.readString();
        date = buf.readInt();
        if (date < 0)
            throw new RuntimeException("Forbidden value on date = " + date + ", it doesn't respect the following condition : date < 0");
    }
    
}
